### PR TITLE
Prac 5 code review request

### PR DESCRIPTION
**Pull request by sachhm**
@ThomasMehes would you be able to just do a quick overview. I think the logic is good, but I'm not 100%. Thanks a lot :smile:

### CODE REVIEW

File: prac_05/emails.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+"""
+CP1404/CP5632 Practical 05
+Emails
+Estimate: 25 minutes
+Actual:  minutes
```
Comment (ThomasMehes) [inline prac_05/emails.py:5]: Must have forgotten to update how long it actually took

File: prac_05/emails.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+"""
+
+
+def main():
+    """Main program to store users emails and names in a dictionary"""
+    emails_to_names = {}
```
Comment (ThomasMehes) [inline prac_05/emails.py:11]: maybe just me, but email_to_names sounds like a function not a library

File: prac_05/emails.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    for email, name in emails_to_names.items():
+        print(f"{name} ({email})")
+
+
+def extract_name(email):
+    """Extract name from a given email"""
+    name = email.split("@")
+    name = name[0]
+    name = name.replace(".", " ")
+
```
Comment (ThomasMehes) [inline prac_05/emails.py:34]: Your code can't handle emails with numbers or any other special characters
could include code like:
characters_to_remove = ".,!?'*$#1234567890"
 name = " ".join(username_parts)
 name = "".join(char for char in name if char not in characters_to_remove)

File: prac_05/hex_colours.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+"""
+CP1404/CP5632 Practical 05
+Look up hexadecimal codes
+"""
+
+NAME_TO_CODE = {"absolutezero": "0048ba", "acidgreen": "#b0bf1a", "aliceblue": "#f0f8ff", "amaranth": "#e52b50",
```
Comment (ThomasMehes) [inline prac_05/hex_colours.py:6]: the same comment as before, I don't find this naming helpful, partially if you consider it was a part of a larger program.

File: prac_05/wimbledon.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
+"""
+CP1404/CP5632 Practical 05
+Wimbledon
+Estimate: 20 minutes
+Actual:  minutes
```
Comment (ThomasMehes) [inline prac_05/wimbledon.py:5]: ...

**Review by ThomasMehes — COMMENTED** [review]
Super sorry about the late review. Overall everything seems really good, I tested your code and all of them work perfectly besides handling emails with special characters and numbers. Everything else is just naming which is subjective anyway. Well done!

File: prac_05/emails.py (hunk 1 above)
Comment (sachhm) [reply prac_05/emails.py:5]: Yep noted. I remember writing it down on paper. Forgot to log it on the comment :facepalm:

File: prac_05/emails.py (hunk 2 above)
Comment (sachhm) [reply prac_05/emails.py:11]: _"A reasonable way to name dictionaries (aka maps) is the key_to_value pattern, e.g., name_to_phone or level_to_reward (using something like phones or levels here would imply they are lists/sets/tuples/etc. not dictionaries)."_

I'm just following the [style guide](https://github.com/CP1404/Starter/wiki/Style-Guide) for this. Maybe renaming it to "email_to_code" would be a better move.

File: prac_05/wimbledon.py (hunk 5 above)
Comment (sachhm) [reply prac_05/wimbledon.py:5]: Yep, its always the small things.  I greatly underestimated this one. Took me nearly an hour to complete.

File: prac_05/hex_colours.py (hunk 4 above)
Comment (sachhm) [reply prac_05/hex_colours.py:6]: Again, I was just following the style guide.

File: prac_05/emails.py (hunk 3 above)
Comment (sachhm) [reply prac_05/emails.py:34]: You're absolutely right here. The sample output didn't seem to account for numbers, so neither did I (Jimbo546).
image

**Comment by sachhm** [comment]
> Super sorry about the late review. Overall everything seems really good, I tested your code and all of them work perfectly besides handling emails with special characters and numbers. Everything else is just naming which is subjective anyway. Well done! 

Thanks for reviewing my work! I appreciate your feedback.

**Pull request merged**